src/solvers/dg_p4est/dg.jl

```diff
@@ -2,11 +2,22 @@
 # It constructs the basic `cache` used throughout the simulation to compute
 # the RHS etc.
 function create_cache(mesh::P4estMesh, equations::AbstractEquations, dg::DG, ::Any, ::Type{uEltype}) where {uEltype<:Real}
-  elements = init_elements(mesh, equations, dg.basis, uEltype)
+  # Make sure to balance the p4est before creating any containers
+  # in case someone has tampered with the p4est after creating the mesh
+  p4est_balance(mesh.p4est, P4EST_CONNECT_FACE, C_NULL)
+  # Due to a bug in p4est, the forest needs to be rebalanced twice sometimes
+  # See https://github.com/cburstedde/p4est/issues/112
```

Comment:
If you haven't done it yet, you should create an issue that references the p4est issue and states that this should be fixed once the upstream bug is fixed.